drivers/hmis/app/models/hmis/form/record_type.rb

```diff
@@ -0,0 +1,81 @@
+module Hmis::Form
+  RecordType = Struct.new(:id, :owner_type, :processor_name, keyword_init: true) do
+    extend Enumerable
+
+    def self.each(&block)
+      all.each(&block)
+    end
+
+    def self.find(id)
+      @by_id ||= all.index_by(&:id)
+      @by_id[id]
+    end
+
+    def self.all
+      @all ||= [
+        new(
+          id: 'ASSESSMENT',
+          owner_type: 'Hmis::Hud::Assessment',
+          processor_name: 'CeAssessment',
+        ),
+        new(
+          id: 'CLIENT',
+          owner_type: 'Hmis::Hud::Client',
+          processor_name: 'Client',
+        ),
+        new(
+          id: 'CURRENT_LIVING_SITUATION',
+          owner_type: 'Hmis::Hud::CurrentLivingSituation',
+          processor_name: 'CurrentLivingSituation',
+        ),
```

Comment:
It feels a little off that we now need a mapping between the **processor name** and the **owner type**, since the processor is supposed to be responsible for "knowing" which record(s) to update (usually by setting `factory_name` which will use the appropriate factory to find/create the record it needs to process..).

Another way could be for the processor to still "own" the knowledge of which record it modifies, but expose that externally as a class method. Looking at the existing (non-class) methods, something in the neighborhood of:
```
processor_name = 'CurrentLivingSituation'
Hmis::Form::FormProcessor.valid_containers[processor_name].record_type
```


I don't know if that's necessarily better, but it does feel more inline with the initial intention of the Form Processor abstraction (though I admit it has gotten away from us).

Am I correct that `PublishFormDefinition` is the only place we currently look at this `owner_type` attribute on this object?

Reply:
Lets keep the currently implementation as discussed. Doesn't seem worth the effort to co-locate the mappings as constants when they wouldn't actually be used during processor evaluation. Thanks!